### ISSUE TITLE
Issue when putting a spreaded tcolorbox inside an enumitem list: size adjustements fail depending on page parity in twosided documents

### ISSUE THREAD

**Issue by v-forman** [T-F-S/tcolorbox#76]
This issue is also described here:
https://tex.stackexchange.com/questions/502194/issue-when-putting-a-tcolorbox-inside-an-enumitem-list-at-page-break-automatic/502227

I want to write a two-sided document with different left an right margins. One of them is larger to accommodate for margin notes.

I also have defined a tcolorbox environment to emphasize some content. The resulting box is supposed to "ignore" the larger margin (as margin notes are not supposed to happen inside) and nearly span the whole page width. For that purpose, I use the spread sidewards key of tcolorbox.

It works fine, except when I put a tcolorbox inside a list item, after loading the enumitem package.

Here is a minimal example demonstrating the problem. As people reported on stackexchange, on all pages with a certain parity (which seems to depend on the package version), itemized tcolorbox are not sidewards spreaded as they should. I also ran some tests with breakable, sidewards spreaded itemized boxes, and the issue is the same.

```latex
\documentclass{article}
\usepackage[a4paper,hmargin={6cm,2cm},twoside]{geometry}
\usepackage{enumitem}
\usepackage{tcolorbox}
\usepackage{lipsum}

\begin{document}
  \begin{itemize}
    \item
    \lipsum[1]
    \begin{tcolorbox}[spread sidewards=-1cm]
      This box should have both its sides at 1cm from the paper sides. It works fine.
    \end{tcolorbox}
    \lipsum[2-6]
    \begin{tcolorbox}[spread sidewards=-1cm]
      This box should also have both its sides at 1cm from the paper sides... but it's wrong!
    \end{tcolorbox}
  \end{itemize}
  \begin{tcolorbox}[spread sidewards=-1cm]
    This box should have both its sides at 1cm from the paper sides. It works fine.
  \end{tcolorbox}
\end{document}
```

I found a dirty fix for this issue, as shown below. It deals with breakable, full-width boxes as explained above.

```latex
\makeatletter
\tcbset{
  spread sidewards/.style={
    width=\paperwidth+(#1)+(#1),
    check odd page,
    show bounding box,
    extras={
      if odd page={
        enlarge right by=-\Gm@rmargin-(#1),
        grow to left by=\@totalleftmargin+\Gm@lmargin+(#1),
      }{
        enlarge right by=-\Gm@lmargin-(#1),
        grow to left by=\@totalleftmargin+\Gm@rmargin+(#1),
      }
    },
  },
  spread sidewards/.default=\z@,
}
```

[![Working box][1]][1]
[![Faulty box][2]][2]

  [1]: https://i.stack.imgur.com/2pnmx.png
  [2]: https://i.stack.imgur.com/FqS4s.png

**Comment by T-F-S** [comment]
Thank you for reporting this issue. I will correct it for the next version.

The following patch should work for the meantime:

```latex
\makeatletter
\tcbset{%
  spread inwards/.style={%
    if odd page={grow to left by=1in+\hoffset+\oddsidemargin+\@totalleftmargin+(#1)}%
      {grow to right by=\paperwidth-\textwidth-1in-\hoffset-\evensidemargin+(#1)}%
  },
  spread outwards/.style={%
    if odd page={grow to right by=\paperwidth-\textwidth-1in-\hoffset-\oddsidemargin+(#1)}%
      {grow to left by=1in+\hoffset+\evensidemargin+\@totalleftmargin+(#1)}%
  },
}
\makeatother

```

**Comment by T-F-S** [comment]
Resolved for version 4.21 (2019-09-19)
